### PR TITLE
Build test targets for all server platforms

### PR DESCRIPTION
**Pull request by luxas**
**What this PR does / why we need it**:

:facepalm:

I really should have checked this before code freeze, but tbh forgot it in the rush. Also I thought this was the case already...
As part of https://github.com/kubernetes/features/issues/288; these binaries should be built for all server platforms indeed.

This is just a straightforward add to that list.
Can we please get this into v1.8?
There is virtually no risk involved here really...

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Build test targets for all server platforms
```
@ixdy @jdumars @mkumatag

### CODE REVIEW

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/testserver/resources.go
Comment (mkumatag) [inline staging/src/k8s.io/apiextensions-apiserver/test/integration/testserver/resources.go]: why this change part of this PR?

**Review by mkumatag — APPROVED** [review]
Have a minor comment otherwise lgtm.

File: staging/src/k8s.io/apiextensions-apiserver/test/integration/testserver/resources.go
Comment (luxas) [reply staging/src/k8s.io/apiextensions-apiserver/test/integration/testserver/resources.go]: otherwise arm 32 bit fails due to that the constant is larger than int32 (default on arm) can handle

Comment (mkumatag) [reply staging/src/k8s.io/apiextensions-apiserver/test/integration/testserver/resources.go]: okay

**Comment by ixdy** [comment]
/lgtm

though we should probably start breaking up `kubernetes-test.tar.gz` into arch-specific tarballs, since it is getting rather large.

**Comment by ixdy** [comment]
(also note you only need to build the test binaries for architectures you're launching tests from - e.g. if I have a cluster running on arm, but my workstation is amd64, I don't need to build the test binaries for arm.)

**Comment by luxas** [comment]
> (also note you only need to build the test binaries for architectures you're launching tests from - e.g. if I have a cluster running on arm, but my workstation is amd64, I don't need to build the test binaries for arm.)

@ixdy Yeah, I know, but this is required as we're starting to run e2es on Packet ARM servers; where the e2e binaries are running in-cluster => cross-built e2e binaries are needed.

> though we should probably start breaking up kubernetes-test.tar.gz into arch-specific tarballs, since it is getting rather large.

Can you please open an issue with more info about that so we can target it for v1.9?

File: hack/lib/golang.sh
Comment (sttts) [inline hack/lib/golang.sh]: will this cause any build performance regressions?

Comment (luxas) [reply hack/lib/golang.sh]: no, this is for release (and the post-ci, independently running kube-cross job) code only. All other jobs etc perform the "fast path" and only build for amd64

**Comment by luxas** [comment]
Adding the approved label as the release team and @ixdy were happy with the changes here, and we want to get this in for tomorrow's beta.1

**Comment by jdumars** [comment]
@luxas could you please assign a SIG label to this?  Thanks!

**Comment by luxas** [comment]
@jdumars added

**Pull request merged**
